src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,439 @@
+    # Geographic coordinate system - likely LatLon based
+    return LatLon{WGS84Latest}
+  elseif startswith(definition_upper, "PROJCS") || startswith(definition_upper, "PROJCRS")
+    # Projected coordinate system - likely Cartesian based
+    return Cartesian{WGS84Latest}
+  else
+    # Unknown format, default to WGS84
+    @warn "Unrecognized CRS definition format for SRID $srid, defaulting to WGS84"
+    return LatLon{WGS84Latest}
+  end
+end
+
+function gpkgreadattribs(db::SQLite.DB, table_name::String, geom_column::String)
+  # Get all data from the table  
+  data_query = "SELECT * FROM \"$table_name\""
+  data_result = DBInterface.execute(db, data_query)
+  
+  table_data = []
+  for row in data_result
+    # Skip rows with missing geometry
+    blob = getproperty(row, Symbol(geom_column))
+    if !ismissing(blob) && !isnothing(blob)
+      # Create attribute row without geometry column
+      row_dict = Dict()
+      for name in propertynames(row)
+        if name != Symbol(geom_column)
+          row_dict[name] = getproperty(row, name)
+        end
+      end
+      push!(table_data, (; row_dict...))
+    end
+  end
+  
+  # Handle case with no attribute columns (only geometry)
+  if !isempty(table_data) && isempty(first(table_data))
+    return nothing
+  else
+    return Tables.rowtable(table_data)
+  end
+end
+
+function gpkgreadgeoms(db::SQLite.DB, table_name::String, geom_column::String, crs)
+  # Get all data from the table
+  data_query = "SELECT \"$geom_column\" FROM \"$table_name\""
+  data_result = DBInterface.execute(db, data_query)
+  
+  geometries = Geometry[]
+  for row in data_result
+    blob = getproperty(row, Symbol(geom_column))
+    if !ismissing(blob) && !isnothing(blob)
+      geom, _ = parse_gpb(blob, crs)
+      if !isnothing(geom)
+        push!(geometries, geom)
+      end
+    end
+  end
+  
+  return geometries
+end
+
+function gpkgread(fname::String; layer::Int=1, kwargs...)
+  db = SQLite.DB(fname)
+  
+  # Get feature table names
+  contents_query = """
+    SELECT table_name, data_type
+    FROM gpkg_contents
+    WHERE data_type = 'features'
+    ORDER BY table_name
+  """
+  contents_result = DBInterface.execute(db, contents_query)
+  contents = [row.table_name for row in contents_result]
+  
+  if isempty(contents)
+    close(db)
+    error("No vector feature tables found in GeoPackage")
+  end
+  
+  if layer > length(contents)
+    close(db)
+    error("Layer $layer not found. GeoPackage contains $(length(contents)) feature layers")
+  end
+  
+  table_name = contents[layer]
+  
+  # Get geometry column information
+  geom_query = """
+    SELECT column_name, srs_id
+    FROM gpkg_geometry_columns
+    WHERE table_name = ?
+  """
+  geom_result = DBInterface.execute(db, geom_query, [table_name])
+  geom_info = first(geom_result)
+  geom_column = geom_info.column_name
+  srs_id = Int32(geom_info.srs_id)
+  crs = get_crs_from_srid(db, srs_id)
```

Comment:
Shouldn't this second query be encapsulated in the `gpkgreadgeoms` function since it is only used therein?

Reply:
Nevermind. I see you are just extracting the column name.